core/src/main/java/nu/yona/server/analysis/service/ActivityService.java

```diff
@@ -153,22 +163,23 @@ private void createInactivityEntities(Set<IntervalInactivityDto> missingInactivi
 		Map<ZonedDateTime, Set<WeekActivityDto>> weekActivityDtosByZonedDate = mapWeekActivitiesToDtos(
 				weekActivityEntitiesByZonedDate);
 		addMissingInactivity(userAnonymized.getGoals(), weekActivityDtosByZonedDate, interval, ChronoUnit.WEEKS, userAnonymized,
-				(goal, startOfWeek) -> createAndSaveWeekInactivity(userAnonymized, goal, startOfWeek, LevelOfDetail.WEEK_OVERVIEW,
-						missingInactivities),
-				(g, wa) -> createAndSaveInactivityDays(userAnonymized,
+				(goal, startOfWeek) -> createAndSaveWeekInactivity(userAnonymized, earliestPossibleDate, goal, startOfWeek,
+						LevelOfDetail.WEEK_OVERVIEW, missingInactivities),
+				(g, wa) -> createAndSaveInactivityDays(userAnonymized, earliestPossibleDate,
 						userAnonymized.getGoalsForActivityCategory(g.getActivityCategory()), wa, missingInactivities));
 		return weekActivityDtosByZonedDate.entrySet().stream().sorted((e1, e2) -> e2.getKey().compareTo(e1.getKey()))
 				.map(e -> WeekActivityOverviewDto.createInstance(e.getKey(), e.getValue())).collect(Collectors.toList());
 	}
 
-	private WeekActivityOverviewDto getWeekActivityOverview(UUID userAnonymizedId, LocalDate date,
+	private WeekActivityOverviewDto getWeekActivityOverview(UUID userAnonymizedId, LocalDate earliestPossibleDate, LocalDate date,
 			Set<IntervalInactivityDto> missingInactivities)
 	{
-		UserAnonymizedDto userAnonymized = userAnonymizedService.getUserAnonymized(userAnonymizedId);
 		Interval interval = Interval.createWeekInterval(date);
+		assertDateNotTooEarly(earliestPossibleDate, interval.endDate);
```

Comment:
Isn't it more logical to swap the arguments? We are asserting something about the `interval.endDate` here. Even better for readability is maybe `interval.assertNotBefore(earliestPossibleDate)`.

Reply:
The order seems a matter of taste to me, but JUnit also puts the tested value before the expectation, so I'll swap them. Moving it to ``Interval`` doesn't sound logical, as this method is used for other dates than interval boundaries and the method would need to indicate whether it assesses the start or end date.